app/scripts/metamask-controller.js

```diff
@@ -493,11 +494,31 @@ export default class MetamaskController extends EventEmitter {
     this.selectedNetworkController = new SelectedNetworkController({
       messenger: this.controllerMessenger.getRestricted({
         name: 'SelectedNetworkController',
-        allowedActions: ['NetworkController:getNetworkClientById'],
+        allowedActions: [
+          'NetworkController:getNetworkClientById',
+          'NetworkController:getState',
+          `PermissionController:hasPermissions`,
+        ],
         allowedEvents: ['NetworkController:stateChange'],
       }),
+      state: initState.SelectedNetworkController,
     });
 
+    // If the perDomainNetwork feature flag is enabled, use metamask origin provider and block tracker
+    // as the proxy passed to the controllers instantiated below, otherwise use the network controller's provider and block tracker
+    // let networkClient;
+    // if (this.selectedNetworkController.state.perDomainNetwork) {
+    //   networkClient =
+    //     this.selectedNetworkController.getProviderAndBlockTracker(
+    //       METAMASK_DOMAIN,
+    //     );
+    // } else {
+    // }
```

Comment:
for testing purposes due to https://github.com/MetaMask/metamask-extension/issues/22937 / https://github.com/MetaMask/metamask-extension/pull/22964